### ISSUE TITLE
Breadcrumb component not accessible to screen readers

### ISSUE THREAD

**Issue by mustaphaboudouch** [mantinedev/mantine#4159]
### What package has an issue

@mantine/core

### Describe the bug

I don't know if i miss something, but the current implementation of the breadcrumb component is not accessible. When rendered, the resulting code is not readable by screen readers :

```html
<div>
    <a href="#">Link 1</a>
    <div>/</div>
    <a href="#">Link 2</a>
    <div>/</div>
    <a href="#">Link 3</a>
</div>
```

### What version of @mantine/hooks page do you have in package.json?

6.0.9

### Do you know how to fix the issue

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

### Possible fix

To improve the accessibility of the breadcrumb component, I suggest using the following code:

```html
<nav aria-label="breadcrumb">
    <ol>
        <li><a href="#">Link 1</a></li>
        <li role="presentation" aria-hidden="true">/</li>
	<li><a href="#">Link 2</a></li>
	<li role="presentation" aria-hidden="true">/</li>
	<li aria-current="page">Link 3</li>
    </ol>
</nav>
```

- `<ol>` : to convey the order of the breadcrumb links to the browser
- `role="presentation"` : to indicate to the browser that divider has no semantic meaning and is purely for visual purposes
- `aria-hidden="true"` : to remove divider from accessibility tree
- `aria-current="page"` : to indicate to the browser that it is the current page (it shouldn't be clickable since it's the current page)

**Comment by rtivital** [comment]
Screen reader works perfectly fine for me – it reads all links in correct order.

https://user-images.githubusercontent.com/10353856/235200662-7556c1ff-f0db-4146-b3fc-1b2dd2cffcd7.mov

**Comment by mustaphaboudouch** [comment]
Yes, the screen reader can read links, but it reads them as separate links and not as a list of links.
So for the screen reader, these two blocks are similar :
```html
<a href="#">Link 1</a>
<a href="#">Link 2</a>
<a href="#">Link 3</a>
```
```html
<div>
    <a href="#">Link 1</a>
    <div>/</div>
    <a href="#">Link 2</a>
    <div>/</div>
    <a href="#">Link 3</a>
</div>
```
It also does not read the last link as the current page.
Here's a simple implementation that you can use with a screen reader to see the difference : https://codepen.io/mustapha_boudouch/pen/WNaEeNj

**Comment by rtivital** [comment]
You can do that on your side (based on your code). 

```tsx
<Breadcrubms>
  <a href="#">Regular link</a>
  <a href="#" aria-current="page">Current link</a>
</Breadcrubms>
```

You are also assuming that the last link is **current** which may not be the case for some applications.
